src/esp/physics/RigidObject.h

```diff
@@ -0,0 +1,121 @@
+#include <Magnum/DebugTools/ForceRenderer.h>
+#include "esp/assets/Asset.h"
+#include "esp/assets/Attributes.h"
+#include "esp/assets/BaseMesh.h"
+#include "esp/assets/FRLInstanceMeshData.h"
+#include "esp/assets/GenericInstanceMeshData.h"
+#include "esp/assets/MeshData.h"
+#include "esp/core/esp.h"
+#include "esp/scene/SceneNode.h"
+
+namespace esp {
+namespace physics {
+
+enum MotionType { STATIC, KINEMATIC, DYNAMIC };
+
+class RigidObject : public scene::SceneNode {
+ public:
+  RigidObject(scene::SceneNode* parent);
+
+  // TODO (JH) Currently a RigidObject is either a scene
+  // or an object, but cannot be both (tracked by _isScene/_isObject_)
+  // there is probably a better way to abstract this
+  virtual bool initializeScene(
+      assets::PhysicsSceneAttributes& physicsSceneAttributes,
+      std::vector<assets::CollisionMeshData>& meshGroup);
+
+  virtual bool initializeObject(
+      assets::PhysicsObjectAttributes& physicsObjectAttributes,
+      std::vector<assets::CollisionMeshData>& meshGroup);
+
+  ~RigidObject();
+
+  //! Check whether object is being actively simulated, or sleeping
+  virtual bool isActive();
+  virtual void setActive(){};
+
+  //! Force interaction
+  virtual void applyForce(Magnum::Vector3& force, Magnum::Vector3& relPos);
+  // Impulse interaction
+  virtual void applyImpulse(Magnum::Vector3& impulse, Magnum::Vector3& relPos);
+
+  //! (Prototype) For visualizing & debugging
+  void debugForce(Magnum::SceneGraph::DrawableGroup3D& debugDrawables);
+  //! (Prototype) For visualizing & debugging
+  void setDebugForce(Magnum::Vector3& force);
+
+  virtual bool removeObject();
+
+  // ==== Transformations ===
+  //! Need to overwrite a bunch of functions to update physical states
+  virtual SceneNode& setTransformation(
+      const Magnum::Math::Matrix4<float>& transformation);
+  virtual SceneNode& setTranslation(const Magnum::Math::Vector3<float>& vector);
+  virtual SceneNode& setRotation(
+      const Magnum::Math::Quaternion<float>& quaternion);
+
+  virtual SceneNode& resetTransformation();
+  virtual SceneNode& translate(const Magnum::Math::Vector3<float>& vector);
+  virtual SceneNode& translateLocal(const Magnum::Math::Vector3<float>& vector);
+
+  virtual SceneNode& rotate(const Magnum::Math::Rad<float> angleInRad,
+                            const Magnum::Math::Vector3<float>& normalizedAxis);
+  virtual SceneNode& rotateLocal(
+      const Magnum::Math::Rad<float> angleInRad,
+      const Magnum::Math::Vector3<float>& normalizedAxis);
+
+  virtual SceneNode& rotateX(const Magnum::Math::Rad<float> angleInRad);
+  virtual SceneNode& rotateY(const Magnum::Math::Rad<float> angleInRad);
+  virtual SceneNode& rotateZ(const Magnum::Math::Rad<float> angleInRad);
+  virtual SceneNode& rotateXLocal(const Magnum::Math::Rad<float> angleInRad);
+  virtual SceneNode& rotateYLocal(const Magnum::Math::Rad<float> angleInRad);
+  virtual SceneNode& rotateZLocal(const Magnum::Math::Rad<float> angleInRad);
```

Comment:
For all `Magnum::Math::Type<float>` you can say just `Magnum::Type` (and for `double` it would be `Magnum::Typed`).